### ISSUE TITLE
Replace Transifex

### ISSUE THREAD

**Issue by comradekingu** [openstreetmap/iD#7508]
Following the logic of OSM and iD being a place free of tracking not by happen-chance,
it would be great if the translation platform followed suit. To me that is one of the central points of what is and should be the greater OSM ecosystem.

Transifex is not libre software, so it is a bad match all together.

As a translator I find Transifex infuriating, and I can't even log in to contribute to the OSM resource, as doing so would grant Transifex a license to use all my strings for their closed translation memory, which is an updated policy requirement they instated the day EU-GDPR went into effect.

Transifex also doesn't let anyone browse strings freely, so the potential for malice is greater. I'd link to the egregious terms and conditions if the whole site wasn't down right now. Requiring contributors to be over 18 is another detriment to its utility as a platform for OSM.

My recommendation is to use [Weblate](https://weblate.org/), either in as a self-hosted instance, or Hosted Weblate. https://weblate.org/hosting/
https://hosted.weblate.org/legal/ is a far cry from the likes of Transifex and Crowdin, but any platform that can compare itself to Weblate is good with me.

Edit: Full disclosure, I have contributed language fixes to Weblate, and other libre software translation platforms. I play a small role in OSM-NO. I have used all platforms extensively.

**Comment by 1ec5** [comment]
Thanks for starting this conversation. At the time this project chose Transifex, that platform was more open (open source too), and the obvious alternatives had significant downsides: #909. For example, Translatewiki.net required access to push changes to the repository automatically. (It looks like [that’s an option with Weblate](https://docs.weblate.org/en/latest/faq.html#what-are-the-options-for-pushing-changes-back-upstream) but there are alternative workflows.)

For what it’s worth, iD currently uses Transifex for free by virtue of being an open source project. Functionally speaking, the Transifex features that matter most to this project would include:

* Translation memory (since some terms are repeated many times across presets and across the community index)
* Maintainer-provided context, support for a decent translation format (YAML is OK but not the only possibility)
* Access to a community of translators who are familiar with translating OSM software (which we currently get by being part of the OpenStreetMap organization at Transifex)

(In my opinion, built-in plural/grammar support would be nice too.)

While we do have a copy of all the current translations, simply hooking the repository up to a different service would lose history and translation memory, so hopefully there’s a way to export that from Transifex. More importantly, the existing translators would need to migrate with us, so there would need to be ample communication.

Ultimately, switching translation management platforms would require a lot of work on the part of this project’s maintainers as well as disruption for translators, so we should be deliberate about it, even though other platforms are more attractive from a privacy and licensing standpoint.

**Comment by maro-21** [comment]
As an active translator of iD and other OSM projects, I don't agree. For me it's a very good tool.
Weblate doesn't have as many features as Transifex does.
The only disadvantage I found during my 1-year work on translations is missing plural forms for languges with many plural forms. But it's nothing because for iD there is maybe one string that needs it. Maybe it supports plurals somehow but I don't know.

Not only iD is on Transifex but many other OSM projects. It's good to have them in one place.

> Transifex also doesn't let anyone browse strings freely

What do you mean? You can browse them.
And if you want to have them all in one file, you can find them on GitHub.

**Comment by comradekingu** [comment]
@1ec5 Transifex doesn't handle fuzzy strings, which is a pain and a half. It also adds all deleted translations to suggestions, so if one sees an error made (which already is clunky), it is clunky to navigate to, and just wastes time. The best libre software translators are already on Hosted Weblate, and other Weblate instances.
Weblate has support for a lot more OAuth-2.0 logins, so signing up is easier.
There is no way of seeing many languages at the same time, can't even see a language other than the one-to-one without disguising the source, and very very few few translators even know where that is hidden in the settings, try to find it. Weblate has matrix and Zen modes, help-languages, and even all languages for a given string.

@maro-21 Without logging in, https://www.transifex.com/openstreetmap/id-editor/ is as far in you get.
Transifex also has a horrid user-role management system, meaning it is too cumbersome to fix errors in other languages. Speaking of, there is no way to see failing checks, much less for the whole project, or organization. On Weblate one can do it on the whole instance, and it is one of the things that prevent malice.

Finding things via Git or on GitHub is not what translators generally do, and they generally use the translation platform. Transifex doesn't have any links to source, no direct editing of source language, a non-threaded messaging system, and a bug-reporting system that is extremely taxing to work with.
There is no connection between upstream TX and TX even, and their paid support is famously bad.

On TX, OSM is _one_ organization, that is not much of an economy of scale. On TX it has very little relevance that things are even in the same organization. If your goal is to have things in one place, if OSM adds itself to Hosted Weblate, it will benefit from a much better and bigger translation memory, and enjoy the company of:

https://hosted.weblate.org/projects/osmand/
https://hosted.weblate.org/projects/getback_gps/
https://hosted.weblate.org/projects/routeconverter/
https://hosted.weblate.org/projects/openorienteering/
https://hosted.weblate.org/projects/trackbook/#information
https://hosted.weblate.org/projects/ghini/
https://hosted.weblate.org/projects/cityzen/#information
https://hosted.weblate.org/projects/min-cal-widget/#information
https://hosted.weblate.org/projects/trustroots/

Reconsidering the rationale of moving to TX in what is now 2020:

> The launchpad translation system is good, but it only supports the gettext format which would mean switching. Translatewiki isn't as turnkey as would be ideal, so I looked at a couple options.
> 
> [Transifex](https://www.transifex.com) is great. Its open-source,

Not anymore. Translatewiki got better, and has a nice system of sharing available manpower onto smaller projects. Sadly it added Google Analytics, so out of the running in terms of what is on offer. Weblate became what Pootle was, but is simpler to use and navigate. Launchpad suffered functionality problems due to servers not responding outside of a few hours of low demand.
Transifex was not the best tool of the day, and it is so far down the list today one struggles to find anything worse, Crowdin is its nearest rival for spying and plain misguided functionality. Having used most every platform and tool extensively before and since then, I can only say Transifex is garbage.
I also read all the terms and conditions, cookie policy, all privacy policies and so on for every platform.

>widely used,

Libre software projects and organizations are leaving it.

>free

Also changed. Besides, Transifex likes to hold projects and organizations over the barrel once the gain enough size. Their prices are insane. 6 figure prices insane.

Weblate OTOH can be used freely as a platform, and using it as a service is actually gratis for libre software. Contributing to a closed TX translation memory and being monetized with tracked advertising is not "free". The gratis Hosted Weblate hosting also features contextual screenshots, grouping, and descriptions one can actually find.

> well documented

https://docs.weblate.org/en/latest/ I find a lot more extensive. Also Weblate itself is more than a web platform as-a-service with an API. You won't find a lot of documentation on the [tracking and spying](https://www.transifex.com/legal/cookies-policy/) TX [does](https://www.transifex.com/legal/privacy/), whereas the legal documents for Weblate are actually [translated on Hosted Weblate](https://hosted.weblate.org/projects/weblate/legal/).

> well designed.

Not for translations, not for admins, not for managers, not for vision impaired/blind people, etc. it has also stood nearly completely still, only has one clunky editor mode, and is full to the brim of bugs and bad design choices.

 > It doesn't support json though.

[Weblate does](https://docs.weblate.org/en/latest/formats.html#json-files).

> We're now using Transifex. Its got its annoying parts
https://github.com/openstreetmap/iD/issues?page=1&q=is%3Aissue+transifex
still does, probably always will.

TL;DR If one champions one should use OSM over the alternatives, OSM really should move away from what presents itself as a mirror image of why OSM is so good—a decision which will even improve matters.

**Comment by quincylvania** [comment]
I'd theoretically be okay with switching iD to a better platform but I'll need to spend time looking into all the practical matters.

**Comment by manfredbrandl** [comment]
From #7870 : Transifex does remove a translation when the source string changes. Instead of changing the translation one has to find the translated text somewhere else and enter it again before changing. This is cumbersome when minor changes (e.g. spelling, wording, small additions) in the source strings occur.

**Comment by comradekingu** [comment]
@quincylvania https://docs.weblate.org/en/latest/
You can now set it up yourself on https://hosted.weblate.org/hosting/ and then just request continued hosting once that is done.
I can help set it up if you want.

**Comment by comradekingu** [comment]
@behrisch 
No, no, and no.
Edit: Hosted Weblate is not the same as Weblate the software, but it doesn't limit the amount of languages or strings.
It is data they are paying for, so there is a practical limit there.
There are other alternatives like https://weblate.bubu1.eu/ or self-hosting.

**Comment by behrisch** [comment]
I am sorry the evaluation of our project did not finish yet, hence the limit. I deleted my misleading comment

**Comment by comradekingu** [comment]
@behrisch Setup isn't a particular strength of Weblate. I can help if you add "kingu".

**Comment by Openstreetmapler** [comment]
Hmm … WHY is Transifex **still** used for this project?? I want to help the OSM iD project, but I HATE being FORCED to give away my sensitive data to that company – to support an open-source, privacy-friendly project?? **That** is a paradox! Please migrate to one of the other platforms!
> > [Transifex is] well designed.
>
> Not for translations, not for admins, not for managers, not for vision impaired/blind people, etc. it has also stood nearly completely still, only has one clunky editor mode, and is full to the brim of bugs and bad design choices.

It’s time to do the switch! **C’mon, you can do it! 💪**

**Comment by yellowsoar** [comment]
BTW, Go Map! is on Weblate too.
https://hosted.weblate.org/projects/go-map/

**Comment by andrewharvey** [comment]
Transifex seems to be behaving weirdly with the en_* translations.

Recently it recorded @k-yle as making a translation to en_AU which he says he's never had permission to edit.

Then also I apparently edited en_GB which I don't think I've ever gone into.

Happy to move to another platform so long as it can support a variety of English translations including, US English, en-GB, en-CA, en-AU, en-NZ, en-SG, and more.

**Comment by 1ec5** [comment]
> Happy to move to another platform so long as it can support a variety of English translations including, US English, en-GB, en-CA, en-AU, en-NZ, en-SG, and more.

Unfortunately, that seems to be a sticking point for Translatewiki.net (as seen in openstreetmap/openstreetmap-website#4975). On the other hand, Transifex’s support for language variants is _too_ complete, as we often get unnecessary requests to translate them despite the unqualified language being untranslated or undertranslated.

**Comment by andrewharvey** [comment]
Weblate seems to support custom languages https://docs.weblate.org/en/latest/admin/languages.html

**Comment by k-yle** [comment]
See also: https://github.com/openstreetmap/id-tagging-schema/commit/defe6cb5564d76b6ed4f80aede3db736185e9bf5#commitcomment-153871643 

Since the most recent release, it's impossible to review the changes to `en-*.json`, because those files now transclude every untranslated string from the base language.

**Comment by andrewharvey** [comment]
#11512 is another reason to replace Transifex, they make it confusing for people to create an account with a gmail etc email address to contribute.

**Comment by simonpoole** [comment]
I would like to thank everybody here for discussing this without addressing and inviting the maintainers of the other OSM projects translated on transifex, cooperating with other projects is just so an outlier in OSM space and needs to be fixed by breaking it.

That said the whole reason we moved translations to a common organisation at the time (the only alternative would have been translatewiki which at the time was totally broken and not accepting new projects), was so that we could share translations and translators.

**Comment by 1ec5** [comment]
> I would like to thank everybody here for discussing this without addressing and inviting the maintainers of the other OSM projects translated on transifex, cooperating with other projects is just so an outlier in OSM space and needs to be fixed by breaking it.

Should we discuss any potential move as a broader coordinated effort somewhere else? I’m unfamiliar with the discussions that led to the current arrangement. I figured it happened organically, with projects joining the organization over time. Nowadays any split would be more difficult than the original setup.

> That said the whole reason we moved translations to a common organisation at the time (the only alternative would have been translatewiki which at the time was totally broken and not accepting new projects), was so that we could share translations and translators.

This echoes the requirements I listed in https://github.com/openstreetmap/iD/issues/7508#issuecomment-612468995. In principle a different TMS could meet these needs, but we need to be sure.

There’s a tradeoff between our ability to coordinate among OSM projects and the “economies of scale” mentioned in https://github.com/openstreetmap/iD/issues/7508#issuecomment-612573557. We benefit from having a barrier between OSM and non-OSM projects. As I alluded to in https://github.com/openstreetmap/openstreetmap-website/issues/6641#issuecomment-3683264739, Translatewiki.net’s lack of such a barrier can result in drive-by translations that misunderstand the context or depart from OSM norms, and we have little control over that. At the same time, some languages would indeed benefit from the project having wider exposure.

**Comment by matkoniecz** [comment]
@simonpoole Should we discuss any potential move as a broader coordinated effort somewhere else? If yes, then where?

**Comment by simonpoole** [comment]
Currently there isn't really an established  non-propreitary place to discuss OSM development related topics with a wider audience (I fully understand the irony of writing that here). The only realistic place is likely the forum, iirc there's already a discussion on creating a dev category.
